### ISSUE TITLE
Home directory names receive unexpected characters

### ISSUE THREAD

**Issue by MattF-NSIDC** [CryoInTheCloud/hub-image#62]
My home directory is unexpectedly named `jupyter-mattf-2dnsidc`. I don't have a great theory on where the `2d` comes from, but perhaps coincidentally, that is the urlencoded string for `-`, and it appears right after the `-` in my username. :shrug:

**Comment by yuvipanda** [comment]
Yep, that's jupyterhub naming escaping. '-' is the safe escape character, so that gets escaped too.

**Comment by MattF-NSIDC** [comment]
I'm still confused :) If the `-` is being escaped, why is it still present in my username?

**Comment by yuvipanda** [comment]
> Does this mean you're using `-` where typically a `\` would be the escape character?

Exactly right. https://github.com/minrk/escapism/blob/master/escapism.py#L40 is the package used consistently throughout jupyterhub, and we consistently use '-' rather than '\' or '_' to be DNS Safe, as many parts of kubernetes require DNS safe escaping.

**Comment by MattF-NSIDC** [comment]
Ah! Thanks, that makes much more sense now :)
